pkg/tkctl/readable/print_flags.go

```diff
@@ -49,14 +47,14 @@ func (p *PrintFlags) ToPrinter(withKind, withNamespace bool) (printers.ResourceP
 		}
 		return printer, nil
 	}
-	// Reuse kubectl HumanReadablePrinter
-	printer := kubeprinters.NewHumanReadablePrinter(scheme.Codecs.UniversalDecoder(),
-		kubeprinters.PrintOptions{
-			WithNamespace: withNamespace,
-			Wide:          p.OutputFormat == "wide",
-			WithKind:      withKind,
-		})
-	// Add custom handlers
-	AddHandlers(printer)
-	return printer, nil
+	printer := kubeprinters.NewTablePrinter(kubeprinters.PrintOptions{
+		WithNamespace: withNamespace,
+		Wide:          p.OutputFormat == "wide",
+		WithKind:      withKind,
+	})
+	tableGenerator := kubeprinters.NewTableGenerator().With(AddHandlers)
+	options := kubeprinters.GenerateOptions{
+		Wide: p.OutputFormat == "wide",
+	}
+	return NewLocalPrinter(printer, tableGenerator, options), nil
```

Comment:
in the new code, the printer itself cannot register handlers. `LocalPrinter` is used to generate Table objects locally.